src/xcode/ENA/ENA/Source/Services/OTP/OTPService.swift

```diff
@@ -240,12 +240,17 @@ final class OTPService: OTPServiceProviding {
 			}
 
 			switch result {
-			case .success(let expirationDate):
+			case .success(let otpForSRSResponsePropertiesReceiveModel):
+				guard let expirationDate = otpForSRSResponsePropertiesReceiveModel.expirationDate else {
+					completion(.failure(.restServiceError(.receivedResourceError(.invalidResponseError))))
+					return
+				}
+				
 				// Success: We store the authorized otp with timestamp and return the token.
 				let verifiedToken = OTPToken(
 					token: otp,
 					timestamp: Date(),
-					expirationDate: expirationDate.expirationDate
+					expirationDate: ISO8601DateFormatter.justUTCDateFormatter.date(from: expirationDate)
```

Comment:
It could potentially crash if the formatter will not be able to process the string

Reply:
good point, is guarded